pkg/storage/mvcc_test.go

```diff
@@ -5248,6 +5248,58 @@ func TestResolveIntentWithLowerEpoch(t *testing.T) {
 	}
 }
 
+// TestTimeSeriesMVCCStats ensures that merge operations
+// result in expected increase in timeseries data
+func TestTimeSeriesMVCCStats(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	for _, engineImpl := range mvccEngineImpls {
+		t.Run(engineImpl.name, func(t *testing.T) {
+			engine := engineImpl.create()
+			defer engine.Close()
+			var ms = enginepb.MVCCStats{}
+
+			// Perform the same sequence of merges on the same key
+			// twice and record timeseries data
+			var firstSysBytes = ms.SysBytes
+			var firstLiveBytes = ms.LiveBytes
+			var firstValBytes = ms.ValBytes
```

Comment:
The test as written is a bit misleading. MVCCMerge takes a pointer to this MVCCStats guy, updating it in place. So "first{Sys,Live,Val}Bytes" are actually the values before the first write (in this case it will always be 0, 0, 0. Instead of unpacking it each time, perhaps we could use something like the following instead?

```
ms := &enginepb.MVCCStats{}

// ...
initialMS := *ms // It'll still be zeros before the first write.

// MVCCMerge(..., ms, ...)
firstMS := *ms

// ...
```

And then comparing things like `firstMS.SysBytes` with `secondMS.SysBytes`?

Reply:
Definitely. I was trying to move the variables declarations around to improve readability and overlooked this. Sorry about that!

Reply:
Since you also suggested adding `initialMS`, are you also suggesting I should validate MVCCStats increases correctly for the first merge as well?